posthog/models/person/person.py

```diff
@@ -65,9 +68,15 @@ def split_person(self, main_distinct_id: Optional[str], max_splits: Optional[int
             if not distinct_id == main_distinct_id:
                 with transaction.atomic():
                     pdi = PersonDistinctId.objects.select_for_update().get(person=self, distinct_id=distinct_id)
-                    person = Person.objects.create(team_id=self.team_id)
+                    person, _ = Person.objects.get_or_create(
+                        uuid=uuidFromDistinctId(self.team_id, distinct_id),
+                        team_id=self.team_id,
+                        defaults={
+                            "version": original_person_version + 1,
+                        },
+                    )
                     pdi.person_id = str(person.id)
-                    pdi.version = (pdi.version or 0) + 1
+                    pdi.version = max(person.version, original_person_version + 1)
```

Comment:
```suggestion
                    pdi.version = max(person.version, original_person_version, pdi.version) + 1
```
Sometimes we run the script to override distinct_ids to version = 2500 or whatever after they are re-used after deletion, in which case `pdi.version` is higher than person version values. Alternatively we can keep this as is (`pdi.version = (pdi.version or 0) + 1`), which would also work iiuc.